File: Lab_7/Codes/Erosion.m
% Read the original image
originalImage = imread('baby2.jpg');
% Define a structuring element for erosion
se = strel('disk', 5); % You can adjust the size of the disk
% Initialize an output image
erodedImage = zeros(size(originalImage), 'like', originalImage);
% Perform erosion on each channel
for channel = 1:size(originalImage, 3)
    erodedImage(:, :, channel) = imerode(originalImage(:, :, channel), se);
end
% Create a figure to display the images
figure;
% Display the original image
subplot(1, 2, 1);
imshow(originalImage);
title('Original Image');
% Display the eroded image
subplot(1, 2, 2);
imshow(erodedImage);
title('Eroded Image')

File: Lab_7/Codes/Erosion.py
import cv2
import numpy as np
import matplotlib.pyplot as plt

# Load the image
original_image_rgb = cv2.imread('baby2.jpg')

# Check if the image is loaded properly
if original_image_rgb is None:
    print("Error: Could not read the image. Please check the file path.")
    exit()

# Convert the image from BGR (OpenCV format) to RGB (matplotlib format)
original_image_rgb = cv2.cvtColor(original_image_rgb, cv2.COLOR_BGR2RGB)

# Define the structuring element (disk-shaped) for erosion
se = cv2.getStructuringElement(cv2.MORPH_ELLIPSE, (5, 5))

# Perform erosion on each channel
eroded_image = np.zeros_like(original_image_rgb)
for channel in range(3):
    eroded_image[:, :, channel] = cv2.erode(original_image_rgb[:, :, channel], se)

# Display the images
plt.figure(figsize=(10, 5))
plt.subplot(1, 2, 1)
plt.imshow(original_image_rgb)
plt.title('Original Image')
plt.subplot(1, 2, 2)
plt.imshow(eroded_image)
plt.title('Eroded Image')
plt.show()
